jvm/src/main/java/com/microsoft/bond/protocol/FastBinaryReader.java

```diff
@@ -0,0 +1,147 @@
+package com.microsoft.bond.protocol;
+
+import com.microsoft.bond.BondDataType;
+
+import java.io.InputStream;
+
+public class FastBinaryReader<S extends InputStream> implements TaggedProtocolReader {
+    public final S stream;
```

Comment:
The idea is that you only need to pass the `FastBinaryReader` around if you want to reuse or close your stream somewhere far from where it's created.

I could be talked out of this fairly easily.